pkg/client/round_tripper.go

```diff
@@ -39,13 +39,11 @@ func NewClusterRoundTripper(delegate http.RoundTripper) *ClusterRoundTripper {
 
 func (c *ClusterRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
 	cluster, ok := ClusterFromContext(req.Context())
-	if !ok {
-		return nil, fmt.Errorf("expected cluster in context")
+	if ok {
+		req = req.Clone(req.Context())
+		req.URL.Path = generatePath(req.URL.Path, cluster)
+		req.URL.RawPath = generatePath(req.URL.RawPath, cluster)
```

Comment:
@varshaprasad96 is this to make the controller-runtime-example working in a non-kcp environment - as it is described here? https://github.com/kcp-dev/apimachinery/issues/32

If so, then I already worked on that issue and to be honest, I originally had the same changes in my code as well. However, I realized that the creation of the controller manager has to be done in a two ways - based on the fact if it should be cluster aware or not. Take a look at this PR changes https://github.com/kcp-dev/controller-runtime-example/pull/8/files#diff-2873f79a86c0d8b3335cd7731b0ecf7dd4301eb19a82ef7a1cba7589b5252261R105-R127
This means that this `ClusterRoundTripper` will be used only in kcp environment - which means when the cluster name should be always available.

Reply:
Because of using different:
* cache
* RoundTripper and thus also the client
* MapperProvider

When the kcp API is available, then the manager is created via:
```go
kcp.NewClusterAwareManager(cfg, options)
```
and the kcp API is not available, then it uses the standard function:
```go
ctrl.NewManager(restConfig, options)
```


It was much easier to do this small if-else logic in the controller-runtime-example `main.go` code, than try to make the `ClusterAwareCache`, `ClusterAwareClient`, and `ClusterAwareMapperProvider` generic enough so it would work in both environments.

Reply:
Ok, thanks. Long term, we want to just have a single path in a main.go that supports both modes.

Reply:
Ok, I wasn't aware of that. Thanks a lot for the links @varshaprasad96 :pray: